hypervisor/vm_console.go

```diff
@@ -59,20 +59,19 @@ func watchVmConsole(ctx *VmContext) {
 		return
 	}
 
-	conn, err := utils.UnixSocketConnect(ctx.ConsoleSockName)
+	reader, err := ctx.DCtx.ConsoleReader(ctx)
 	if err != nil {
-		ctx.Log(ERROR, "failed to connected to %s: %v", ctx.ConsoleSockName, err)
+		ctx.Log(ERROR, "failed to get console for %v %v", ctx.Id, err)
 		return
 	}
+	defer reader.Close()
 
-	ctx.Log(TRACE, "connected to %s", ctx.ConsoleSockName)
-
-	tc, err := telnet.NewConn(conn)
+	tc, err := telnet.NewConn(reader)
```

Comment:
It seams that the above code(L62-L69) should all be moved in to DCtx.ConsoleReader() (those who are using ctx.ConsoleSockName).
DCtx.ConsoleReader() for xenpv doesn't returns a telnet socket, you cannot use telnet.NewConn() for it.